### PR TITLE
SDK-552 Add share datasource

### PR DESCRIPTION
**Pull request by cristiGuranIonos**
## What does this fix or implement?

Add share datasource.

## Checklist

- [X] Tests added or updated
- [X] Documentation updated
- [X] Changelog updated and version incremented
- [ ] Github Issue linked if any

### CODE REVIEW

File: ionoscloud/resource_nic_test.go [hunk 1]
```diff
@@ -164,7 +164,7 @@ resource "ionoscloud_nic" "database_nic" {
   lan = 2
   dhcp = false
   firewall_active = false
-  ips            = [ ionoscloud_ipblock.test_server.ips[0], ionoscloud_ipblock.test_server.ips[1] ]
+  ips = [ ionoscloud_ipblock.test_server.ips[0], ionoscloud_ipblock.test_server.ips[1] ]
```
Comment (IuliaBlindu) [inline ionoscloud/resource_nic_test.go:167]: Not in this PR purpose, but here i would also tests (if the field is updatable) if you can empty the ips list or add one more ip

File: ionoscloud/provider.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ func Provider() *schema.Provider {
 			ServerResource:             resourceServer(),
 			VolumeResource:             resourceVolume(),
 			GroupResource:              resourceGroup(),
-			"ionoscloud_share":         resourceShare(),
+			shareResource:              resourceShare(),
```
Comment (IuliaBlindu) [inline ionoscloud/provider.go:74]: I think we should be consistent regarding naming - ShareResource instead of shareResource

**Review by IuliaBlindu — APPROVED** [review]

File: ionoscloud/provider.go (hunk 2 above)
Comment (cristiGuranIonos) [reply ionoscloud/provider.go:74]: That would mean exporting the constant to other packages. The question is, do we want to do that?

File: ionoscloud/resource_nic_test.go (hunk 1 above)
Comment (cristiGuranIonos) [reply ionoscloud/resource_nic_test.go:167]: good point, will add tests for these

File: ionoscloud/provider.go (hunk 2 above)
Comment (cristiGuranIonos) [reply ionoscloud/provider.go:74]: Will change to uppercase, maybe we will need these in other packages
